pkg/csi/service/common/authmanager.go

```diff
@@ -63,58 +73,96 @@ func GetAuthorizationService(ctx context.Context, vc *cnsvsphere.VirtualCenter)
-			rwMutex:                           sync.RWMutex{},
-			vcenter:                           vc,
+			datastoreMapForBlockVolumes: make(map[string]*cnsvsphere.DatastoreInfo),
+			datastoreMapForFileVolumes:  make(map[string]*cnsvsphere.DatastoreInfo),
+			rwMutex:                     sync.RWMutex{},
+			vcenter:                     vc,
 		}
-		authManagerInstance.refreshDatastoreIgnoreMap()
 	})
 
 	log.Info("authorization service initialized")
 	return authManagerInstance, nil
 }
 
-// GetDatastoreIgnoreMapForBlockVolumes returns a DatastoreIgnoreMapForBlockVolumes. This map maps datastore url to datastore info
-// which need to be ignored when creating block volume.
-func (authManager *AuthManager) GetDatastoreIgnoreMapForBlockVolumes(ctx context.Context) map[string]*cnsvsphere.DatastoreInfo {
-	datastoreIgnoreMapForBlockVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
+// GetDatastoreMapForBlockVolumes returns a DatastoreMapForBlockVolumes. This map maps datastore url to datastore info
+// which need to be used when creating block volume.
+func (authManager *AuthManager) GetDatastoreMapForBlockVolumes(ctx context.Context) map[string]*cnsvsphere.DatastoreInfo {
+	datastoreMapForBlockVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
 	authManager.rwMutex.RLock()
 	defer authManager.rwMutex.RUnlock()
-	for dsURL, dsInfo := range authManager.datastoreIgnoreMapForBlockVolumes {
-		datastoreIgnoreMapForBlockVolumes[dsURL] = dsInfo
+	for dsURL, dsInfo := range authManager.datastoreMapForBlockVolumes {
+		datastoreMapForBlockVolumes[dsURL] = dsInfo
+	}
+	return datastoreMapForBlockVolumes
+}
+
+// GetDatastoreMapForFileVolumes returns a DatastoreMapForFileVolumes. This map maps datastore url to datastore info
+// which need to be used when creating file volume.
+func (authManager *AuthManager) GetDatastoreMapForFileVolumes(ctx context.Context) map[string]*cnsvsphere.DatastoreInfo {
+	datastoreMapForFileVolumes := make(map[string]*cnsvsphere.DatastoreInfo)
+	authManager.rwMutex.RLock()
+	defer authManager.rwMutex.RUnlock()
+	for dsURL, dsInfo := range authManager.datastoreMapForFileVolumes {
+		datastoreMapForFileVolumes[dsURL] = dsInfo
+	}
+	return datastoreMapForFileVolumes
+}
+
+// refreshDatastoreMap scans all datastores in vCenter to check privileges, and compute the
+// datastoreMapForBlockVolumes
+func (authManager *AuthManager) refreshDatastoreMapForBlockVolumes() {
+	ctx, log := logger.GetNewContextWithLogger()
+	log.Info("auth manager: refreshDatastoreMapForBlockVolumes is triggered")
+	newDatastoreMapForBlockVolumes, err := GenerateDatastoreMapForBlockVolumes(ctx, authManager.vcenter)
+	if err == nil {
+		authManager.rwMutex.Lock()
+		defer authManager.rwMutex.Unlock()
+		authManager.datastoreMapForBlockVolumes = newDatastoreMapForBlockVolumes
+		log.Debugf("auth manager: datastoreMapForBlockVolumes is updated to %v", newDatastoreMapForBlockVolumes)
+	} else {
```

Comment:
You can drop this else. Can you run make golangci-lint and verify

Reply:
same as above.